pkg/util/instancemetadata/prod.go

```diff
@@ -43,12 +43,12 @@ func newProd(ctx context.Context) (InstanceMetadata, error) {
 		},
 	}
 
-	err := p.populateTenantIDFromMSI(ctx)
+	err := p.populateInstanceMetadata()
 	if err != nil {
 		return nil, err
 	}
 
-	err = p.populateInstanceMetadata()
+	err = p.populateTenantIDFromMSI(ctx)
```

Comment:
The order in which these functions are called has to be reversed, since `populateTenantIDFromMSI` now relies on `p.environment`, which is assigned in `populateInstanceMetadata`.